documentation.md

```diff
@@ -0,0 +1,521 @@
+Two distinct tail pointers are required to dynamically manage the two following cases:
+
+- The pre_tail address is given as input to the payload RAM if there is a successful output handshake and the value at the output has an AXI id corresponding to the AXI id of the considered linked list. It must point to:
+  - The second-to-last element of the linked list if the list contains two reponses or more,
+  - The last element of the linked list if the list contains just one response,
+  - rsp_head if the list contains no responses.
+- The tail address is given as input to the payload RAM in all other cases. This case disjunction prevents an output data from being output twice, and prevents any bandwidth drop at the output. It must point to:
+
+  - The last element of the linked list if the list contains one response or more,
+  - pre_tail if the list contains no responses.
+
+<figure class="image">
+  <img src="https://i.imgur.com/Y7XwICc.png" alt="Simulated memory controller internal overview">
+  <figcaption>Fig: Linked list representation. t: tail, pt: pre_tail, rsp: response head, rsv: reservation head. Arrows between extended cells represent the linked list structure (cells must not have consecutive addresses)</figcaption>
+</figure>
+ 
+ 
+#### Lengths
+
+In addition to the pointers, lengths of sub-segments of linked lists are stored to maintain the state of each linked lists, which is not fully defined by the four pointer and metadata RAMs only:
+
+- _rsv_len_: Holds the number of extended cells that have been reserved but have not received any response yet.
+- _rsp_len_: Holds the number of active extended cells.
+
+<figure class="image">
+  <img src="https://i.imgur.com/gj5IPNo.png" alt="Simulated memory controller internal overview">
+  <figcaption>Fig: Example of rsv_len and rsp_len</figcaption>
+</figure>
+
+Additionally, another length is combinatorically inferred:
+
+- _rsp_len_after_out_: Determines what will be _rsp_len_, considering the release of responses but not the acquisition of new data. This signal is helpful in many regular and corner cases as it helps to manage the latency cycle at the output.
+
+#### Extended cell state
+
+##### Internal counters
+
+For each extended RAM cells, additional memory is dedicated to maintaining the extended RAM cell state:
+
+- _rsv_cnt_: Counts the number of responses that are still awaited in the burst. When reserving an extended cell, this counter is set to the address request burst length. The counter is decremented every time a burst response is transmitted from this extended RAM cell to the requester.
+- _rsp_cnt_: Counts the number of responses that are currently stored in the extended cell. The counter is incremented every time a response is acquired and decremented every time a response is released.
+- _burst_len_: Stores the burst length of the address request corresponding to this extended RAM cell.
+
+The following relationship always holds:
+$$burst\_len \ge rsv\_cnt + rsp\_cnt$$
```

Comment:
This does not render nicely in GItHub MarkDown.

Reply:
Changed, thanks!